src/video/x11/SDL_x11window.c

```diff
@@ -1562,6 +1562,97 @@ X11_SetWindowGammaRamp(_THIS, SDL_Window * window, const Uint16 * ramp)
     return 0;
 }
 
+typedef struct {
+    unsigned char *data;
+    int format, count;
+    Atom type;
+} SDL_x11Prop;
+
+/* Reads property
+   Must call X11_XFree on results
+ */
+static void X11_ReadProperty(SDL_x11Prop *p, Display *disp, Window w, Atom prop)
+{
+    unsigned char *ret=NULL;
+    Atom type;
+    int fmt;
+    unsigned long count;
+    unsigned long bytes_left;
+    int bytes_fetch = 0;
+
+    do {
+        if (ret != 0) X11_XFree(ret);
+        X11_XGetWindowProperty(disp, w, prop, 0, bytes_fetch, False, AnyPropertyType, &type, &fmt, &count, &bytes_left, &ret);
+        bytes_fetch += bytes_left;
+    } while (bytes_left != 0);
+
+    p->data=ret;
+    p->format=fmt;
+    p->count=count;
+    p->type=type;
+}
+
+void*
+X11_GetWindowICCProfile(_THIS, SDL_Window * window, size_t * size)
+{
+    SDL_WindowData *data = (SDL_WindowData *) window->driverdata;
+    Display *display = data->videodata->display;
+    XWindowAttributes attributes;
+    Atom icc_profile_atom;
+    char* icc_atom_string;
+    void* ret_icc_profile_data = NULL;
+    CARD8* icc_profile_data;
+    Atom real_type;
+    int real_format;
+    unsigned long real_nitems;
+    SDL_x11Prop atomProp;
+
+    X11_XGetWindowAttributes(display, X11_IsWindowLegacyFullscreen(_this, window) ? data->fswindow : data->xwindow, &attributes);
+    if (X11_XScreenNumberOfScreen(attributes.screen) > 0) {
+        icc_atom_string = SDL_calloc(1, sizeof("_ICC_PROFILE_") + 12);
```

Comment:
Not sure about SDL style but this seems like a strange thing to calloc, couldn't you very easily stack-allocate room for this? It's not a large allocation.

Reply:
Stack-allocation is unsafe and generally discouraged.

Reply:
Also, I believe as written the function leaks this allocation either way.

Reply:
Memory leak fixed.

Reply:
> Dynamic stack-allocation is unsafe and generally discouraged.

I don't understand. It's not dynamic because the size is already fixed as 'sizeof("_ICC_PROFILE" + 12)". As long as you use `snprintf`, which you do, using a fixed-size stack allocation is as safe as using a fixed-size heap allocation, no?

Reply:
On second thought, I made it a stack allocation.